File: LondonDataServices.IDecide.Portal.Client/src/brokers/apiBroker.patientDecisions.ts
import { PatientDecision } from "../models/patientDecisions/patientDecision";
import ApiBroker from "./apiBroker";

class PatientDecisionBroker {
    relativeDecisionsUrl = '/api/PatientDecision';
    private apiBroker: ApiBroker = new ApiBroker();

    async PostPatientDecisionAsync(decision: PatientDecision, headers?: Record<string, string>) {
        const url = `${this.relativeDecisionsUrl}/PatientDecision`;

            return await this.apiBroker.PostAsync(url, decision, headers)
                .then(() => undefined);
    }

    async PostPatientDecisionNhsLoginAsync(decision: PatientDecision) {
        const url = `${this.relativeDecisionsUrl}/PatientDecisionNhsLogin`;

        return await this.apiBroker.PostAsync(url, decision)
            .then(() => undefined);
    }
}

export default PatientDecisionBroker;

File: LondonDataServices.IDecide.Portal.Server.Tests.Acceptance/Brokers/ApiBroker.PatientDecisions.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System.Threading.Tasks;
using LondonDataServices.IDecide.Portal.Server.Tests.Acceptance.Models.PatientDecisions;

namespace LondonDataServices.IDecide.Portal.Server.Tests.Acceptance.Brokers
{
    public partial class ApiBroker
    {
        private const string patientDecisionsRelativeUrl = "api/patientdecision";

        public async ValueTask<Decision> PostPatientDecisionAsync(Decision decision) =>
            await this.apiFactoryClient.PostContentAsync($"{patientDecisionsRelativeUrl}/PatientDecision", decision);
    }
}
